src/seq_mv/csr_matrix.c

```diff
@@ -82,7 +83,10 @@ hypre_CSRMatrixDestroy( hypre_CSRMatrix *matrix )
       if ( hypre_CSRMatrixOwnsData(matrix) )
       {
          hypre_TFree(hypre_CSRMatrixData(matrix), HYPRE_MEMORY_SHARED);
-         hypre_TFree(hypre_CSRMatrixJ(matrix), HYPRE_MEMORY_SHARED);
+         if (hypre_CSRMatrixJ(matrix))
```

Comment:
I removed some if-statements of this kind, since TFree checks: if (NULL) return; but this is not important